src/encoder/encoderfdkaac.cpp

```diff
@@ -414,7 +410,7 @@ void EncoderFdkAac::processFIFO() {
         // Output (result) Buffer
         AACENC_BufDesc outputBuf;
         outputBuf.numBufs = 1;
-        outputBuf.bufs = (void**)&m_pAacDataBuffer;
+        outputBuf.bufs = reinterpret_cast<void**>(m_pAacDataBuffer.data());
```

Comment:
Good catch! That's why reinterpret_cast is dangerous.

```
outputBuf.bufs = &static_cast<void*>(m_pAacDataBuffer.data());
```

Reply:
So note, `.bufs` datatype is `void**`. This means one is already in the "danger zone" :eyes: .

`static_cast<T>`, (`&static_cast<T>`) only works for datatypes it can "safely" convert to. 
 With `void**` it is not possible to assume "safety". So `static_cast<T>` cannot work in this realm. You need to use `reinterpret_cast<void**>` to make "the magic" happen. :sparkles: :fire:

Reply:
Nevertheless, `m_pAacDataBuffer.data()` is the wrong pointer. The API expects an array of pointers to multiple buffers and not just a pointer to a single buffer, i.e. double indirection!

Reply:
I agree with you. So `unsigned char* m_pAacDataBuffer;` and `.bufs = (void**)&m_pAacDataBuffer;` was how it was declared and assigned previously.
In your opinion which one would be preferable, could you help me decide?
```cpp
// [1]
.bufs = reinterpret_cast<void**>(std::addressof(*std::begin(m_pAacDataBuffer)));
// [2]
.bufs = reinterpret_cast<void**>(std::addressof(*m_pAacDataBuffer.data()));
// [3]
.bufs = reinterpret_cast<void**>(&(*m_pAacDataBuffer.data()));
// [4]
.bufs = reinterpret_cast<void**>(std::addressof(std::begin(m_pAacDataBuffer)[0]));
```

Reply:
How about this safe variant that does not require any explicit casts:

```
void* outputBufs[] = {m_pAacDataBuffer.data()};
outputBuf.bufs = outputBufs;
```

Reply:
This is a deep copy unfortunately.

Reply:
Ya, a double implicit way seems more elegant to solve this one. Will go with this. :smiley_cat:

Reply:
The version with the temporary array also reveals the actual semantics.